app/src/main/java/org/piwigo/ui/main/ImagesItemViewModel.java

```diff
@@ -37,4 +54,32 @@ public String getTitle() {
         return title;
     }
 
+    public int getImageId()
+    {
+        return (imageId);
+    }
+
+    public void onClickDo(View v)
+    {
+        Context ctx = v.getContext();
+        while (ctx instanceof ContextWrapper
+                && !(ctx instanceof AppCompatActivity)) {
+            ctx = ((ContextWrapper)ctx).getBaseContext();
+        }
+
+        //TODO Maybe find a better way to do this (note that AppCompatActivity doesn't work on SDK 16)
+        MainActivity mainActivity = (MainActivity)ctx;
+        if(mainActivity != null)
+        {
+            Bundle bundle = new Bundle();
+            bundle.putSerializable("images", images);
```

Comment:
How should we achieve this then ?

Asking the ImagesRepository for the list of images and get them from the cache again ?

Reply:
yes, why not. They are in memory anyhow, so better to get them fresh from there than the serialize/deserialize them (even though I still don't know the details about that process, but it definitly looks like more than passing a reference around...)

And with the full-caching local repository (of which I dream) I think it would be really the best to get it from there and let this one for example also handle the question of which size and dynamically reloading different sizes...